src/ensemble/basic_ensemble_solve.jl

```diff
@@ -84,27 +84,50 @@ function __solve(prob::AbstractEnsembleProblem,
   num_batches = trajectories ÷ batch_size
   num_batches * batch_size != trajectories && (num_batches += 1)
 
-  u = deepcopy(prob.u_init)
+  function batch_function(I)
+    batch_data = solve_batch(prob,alg,ensemblealg,I,pmap_batch_size;kwargs...)
+  end
+
+  if num_batches == 1 && prob.reduction === DEFAULT_REDUCTION
+    elapsed_time = @elapsed u = batch_function(1:trajectories)
+    if typeof(u) <: Vector{Any}
+      _u = map(i->u[i],1:length(u))

```

Comment:
Maybe
```suggestion
      _u = map(identity, u)
```
? That could (potentially) exploit optimizations for `::typeof(identity)`, and does not require an additional range and function.